packages/devtools_app/lib/src/shared/utils.dart

```diff
@@ -92,3 +93,31 @@ extension VmExtension on VM {
     ].join(' ');
   }
 }
+
+/// Mixin that provides a [controller] from package:provider for a State class.
+///
+/// [initController] must be called from [State.didChangeDependencies]. If
+/// [initController] returns false, [didChangeDependencies]
+mixin ProvidedControllerMixin<T, V extends StatefulWidget> on State<V> {
```

Comment:
Can it happen that one widget will need two controllers? If yes, what should we do with name conflict?

Reply:
It could. For those cases we can either:
a) use `ProvidedControllerMixin` for one of the controllers (the primary one if that applies) and pull the other directly from provider via `Provider.of<ControllerType>(context)`
b) pull both controllers from provider via `Provider.of<ControllerType>(context)` and do not use `ProvidedControllerMixin` at all